### DISCUSSION TITLE
Casting in Qute?

### DISCUSSION THREAD

**Discussion by GregJohnStewart** [quarkusio/quarkus#34116] [Q&A] [answered]
I have a pojo that can be one of several sub-classes. This pojo is held in another object, which holds the superclass form. Qute seems to preemptively be judging that the object doesn't have the method:

```
{#when result.getCheckedOutFor().getType()}
{#is OQM_ENTITY} {! I use type flags to specify sub-types, for Jackson !}
{#interactingEntityRef entityRef=result.getCheckedOutFor().getEntity()}{/interactingEntityRef}
{#is EXT_SYS_USER}
{result.getCheckedOutFor().getName()}
{/when}
```
This produces:

```
- {result.getCheckedOutFor().getEntity()}: Property/method [getEntity()] not found on class [tech.ebp.oqm.lib.core.object.storage.checkout.checkoutFor.@NonNull CheckoutFor] 
```
(`CheckoutFor` is my super class, held in `result`)

Is there an easy way to deal with polymorphism in Qute? Casting in the typical sense doesn't work here. Also tried `instanceof` in an `if` statement, same error.

[a comment by an automated account is not shown]

**Reply by mkouba** [reply, marked as answer]
Ok, so this is a build time validation error. We don't know the runtime type of the object returned from the `getCheckedOutFor()` method at this time. You can try to exclude this particular method from validation by means of [`quarkus.qute.type-check-excludes=org.acme.CheckoutFor.getEntity`](https://quarkus.io/guides/qute-reference#quarkus-qute_quarkus.qute.type-check-excludes). It should resolve just fine at runtime.

**Reply by GregJohnStewart** [reply]
Thanks, that works! I realize this is a hard thing to solve, but this resolution isn't amazing, as we are circumventing checks.

Maybe have a way for `#let` to cast?
